### PR TITLE
[excel] (Events) Article section and samples for the onRowSorted and onColumnSorted events

### PR DESCRIPTION
**Pull request by AlexJerabek**
This PR also updates the event list table in the "Work with Events using the Excel JavaScript API" article.

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[5cb74f2](https://github.com/OfficeDev/office-js-docs-pr/commits/5cb74f2c8d8707ead431a2ec241444d995811fd1)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-tables.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-tables.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-worksheets.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-worksheets.md) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-after-column.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-after-column.png) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-after-row.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-after-row.png) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-before.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-before.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C9%5C9%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201909092331197987-1298%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=PRrneWppKe1DFaAVWdBnH5MeLKGOp80jQ7O1T8%2FCyTs%3D&st=2019-09-09T23%3A27%3A22Z&se=2019-10-10T23%3A32%3A22Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-worksheets.md [hunk 1]
```diff
@@ -295,6 +295,56 @@ function onWorksheetChanged(eventArgs) {
 }
 ```
 
+## Sorting events (preview)
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-worksheets.md]:
```suggestion
## Handle sort-related events (preview)
```

File: docs/excel/excel-add-ins-worksheets.md [hunk 2]
```diff
@@ -295,6 +295,56 @@ function onWorksheetChanged(eventArgs) {
...
+
+> [!NOTE]
+> The APIs for these sort-related events are currently available only in public preview. [!INCLUDE [Information about using preview APIs](../includes/using-excel-preview-apis.md)]
+
+The `onColumnSorted` and `onRowSorted` events indicate when any worksheet data is sorted. These events fire whether the sorting is done through UI controls (like table drop-downs) or programmatically. These events are connected to individual `Worksheet` objects and the workbook's `WorkbookCollection`.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-worksheets.md]:
```suggestion
The `onColumnSorted` and `onRowSorted` events indicate when any worksheet data is sorted. These events are connected to individual `Worksheet` objects and to the workbook's `WorkbookCollection`, and will fire whether the sorting is done manually through the Excel user interface or programmatically.
```

File: docs/excel/excel-add-ins-worksheets.md [hunk 3]
```diff
@@ -295,6 +295,56 @@ function onWorksheetChanged(eventArgs) {
...
+
+> [!NOTE]
+> `onColumnSorted` fires when columns are sorted as the result of a left-to-right sort operation. `onRowSorted` fires when rows are sorted as the result of a top-to-bottom sort operation. Sorting a table using the drop-down on a column header results in an `onRowSorted` event. The event corresponds with what is moving, not what is being considered as the sorting criteria.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-worksheets.md]: The style guide won't load for me right now, but IIRC, "drop-down" isn't quite the right terminology. Suggest you look up the recommended term in the style guide (if it'll load for you).

File: docs/excel/excel-add-ins-worksheets.md [hunk 4]
```diff
@@ -295,6 +295,56 @@ function onWorksheetChanged(eventArgs) {
...
+
+The `onColumnSorted` and `onRowSorted` events provide their callbacks with [WorksheetColumnSortedEventArgs](/javascript/api/excel/excel.worksheetcolumnsortedeventargs) or [WorksheetRowSortedEventArgs](/javascript/api/excel/excel.worksheetrowsortedeventargs), respectively. These give more details about the event. In particular, both `EventArgs` have an `address` property that represents the rows or columns moved as a result of the sort operation. Any cell with sorted content is included, even if that cell's value was not part of the sorting criteria.
+
+The following images show what ranges are returned by the `address` property for sort events. First, here is the sample data before sorting:
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-worksheets.md]:
```suggestion
The following images show the ranges returned by the `address` property for sort events. First, here is the sample data before sorting:
```

File: docs/excel/excel-add-ins-worksheets.md [hunk 5]
```diff
@@ -295,6 +295,56 @@ function onWorksheetChanged(eventArgs) {
...
+
+If a top-to-bottom sort is performed on "**Q1**" (the values in "**B**"), the following highlighted rows are returned by `WorksheetRowSortedEventArgs.address`:
+
+![Table data in Excel after a top-to-bottom sort. The rows that have moved are highlighted.](../images/excel-sort-event-after-row.png)
+
+If a left-to-right sort is performed on "**Quinces**" (the values in "**4**") on the original data, the following highlighted columns are returned by `WorksheetColumnsSortedEventArgs.address`:
+
+![Table data in Excel after a left-to-right sort. The columns that have moved are highlighted.](../images/excel-sort-event-after-column.png)
+
+The following code sample shows how to add an event handler to `Worksheet.onRowSorted`. The handler's callback clears the fill color for the range, then fills the cells of the moved rows.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-worksheets.md]:
```suggestion
The following code sample shows how to register an event handler for the `Worksheet.onRowSorted` event. The handler's callback clears the fill color for the range, then fills the cells of the moved rows.
```

File: docs/excel/excel-add-ins-worksheets.md [hunk 6]
```diff
@@ -295,6 +295,56 @@ function onWorksheetChanged(eventArgs) {
...
+
+```js
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-worksheets.md]: Should the "fat arrow" syntax be used in this (JavaScript) code sample, or should the longer `function (...)` syntax be used instead?

**Comment by o365devx** [comment]
Docs Build status updates of commit _[7341ca4](https://github.com/OfficeDev/office-js-docs-pr/commits/7341ca4e9ca8d96855a3e0e4f8408a401f0a1286)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-tables.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-tables.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-worksheets.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-worksheets.md) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-after-column.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-after-column.png) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-after-row.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-after-row.png) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-before.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-before.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C9%5C10%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201909101730110431-1298%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=56lPnKn6tskWn4GE4Q872F7idM%2BwznjEKefiSb7Yaj0%3D&st=2019-09-10T17%3A26%3A18Z&se=2019-10-11T17%3A31%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-worksheets.md (hunk 1 above)
Comment (AlexJerabek) [reply docs/excel/excel-add-ins-worksheets.md]: @kbrandl, How do you feel about calling this "Handle sorting events"? The "related" part makes the events seem distant to the actual sorting.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[7dd33a4](https://github.com/OfficeDev/office-js-docs-pr/commits/7dd33a4f8d8ba4a6d90b1979938799ce22e3b51f)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-tables.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-tables.md) | :warning:Warning | | [Details](#user-content-f1b09002-91463079)
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-worksheets.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-worksheets.md) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-after-column.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-after-column.png) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-after-row.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-after-row.png) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-before.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-before.png) | :white_check_mark:Succeeded | |

<a id="f1b09002-91463079"></a>
### [docs/excel/excel-add-ins-tables.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-tables.md)
 - **Line 283**: **[Warning]** `````Invalid link: '<a href="excel/excel-add-ins-worksheets.md#handle-sort-related-events">Handle sort-related events (preview)</a>'. The file excel/excel-add-ins-worksheets.md doesn't contain a bookmark named 'handle-sort-related-events'.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C9%5C10%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201909101730495983-1298%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=LwxQNkHTYcgM7d8SmCqcIeIl1%2BNjRYzMHjw0ocJSDGQ%3D&st=2019-09-10T17%3A27%3A25Z&se=2019-10-11T17%3A32%3A25Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-worksheets.md (hunk 1 above)
Comment (kbrandl) [reply docs/excel/excel-add-ins-worksheets.md]: That sounds good to me! Be sure to update the hyperlink (text and URL) in the other article that points to this section.

File: docs/excel/excel-add-ins-worksheets.md (hunk 3 above)
Comment (AlexJerabek) [reply docs/excel/excel-add-ins-worksheets.md]: I double-checked and drop-down should be okay for developers.

image

Comment (AlexJerabek) [reply docs/excel/excel-add-ins-worksheets.md]: Though in re-reading that, I'll need to clarify the control type. I'll figure out how best to describe that.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[163f85c](https://github.com/OfficeDev/office-js-docs-pr/commits/163f85c633c5f0837beed96199878c01a7809739)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-tables.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-tables.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-worksheets.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/excel/excel-add-ins-worksheets.md) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-after-column.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-after-column.png) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-after-row.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-after-row.png) | :white_check_mark:Succeeded | |
[docs/images/excel-sort-event-before.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Sort/docs/images/excel-sort-event-before.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C9%5C10%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201909101759307282-1298%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=22yR2nGJ1GbeESk1awYLoccPku8wa54Ho%2B%2B6TkxCTtU%3D&st=2019-09-10T17%3A55%3A32Z&se=2019-10-11T18%3A00%3A32Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-worksheets.md (hunk 3 above)
Comment (ElizabethSamuel-MSFT) [reply docs/excel/excel-add-ins-worksheets.md]: I think if you choose to use that term, then it might be "dropdown"

Comment (AlexJerabek) [reply docs/excel/excel-add-ins-worksheets.md]: I think the best way to describe that Excel control is as a "drop-down menu", like in the above screenshot.

**Pull request merged**